meeting-notes/2018/2018-11-19--go-core-dev-team-weekly.md

```diff
@@ -0,0 +1,197 @@
+  - @hannahhoward
+  
+- **Recording:** _add link to recording once it's online_
+
+## Agenda
+
+- Ask everyone to put their name into the list of attendees
+- Announcements
+- @magik6k's update on datstore benchmarking
+- Round of updates
+  - High-priority updates. Are you blocked? On what and whose help do you need.
+- Ask for general questions. Could be things like:
+  - I'm stuck with something, I don't know who to ask. Who knows who to ask?
+  - Who can help me with xyz?
+
+
+## Notes
+
+#### Announcements
+
+- OKRs still need projections from some team members: https://docs.google.com/spreadsheets/d/139lROP7-Ee4M4S7A_IO4iIgSgugYm7dct620LYnalII/edit?ts=5bef213d#gid=1720716278. Check the spreadsheet for your OKR items and review the mid-Q completion estimate, and provide an estimate for completion by the end of the quarter.
+
+- Weekly sync changes; make time each week for reviewing progress of the three critical go-ipfs efforts; data transport, providing, and data management.
+  - going to center this meeting more around our 3 main endeavors (starting with datastore)
+  
+
+- Reflection issue - improving how we work: https://github.com/ipfs/go-ipfs/issues/5781. Read it over and please add your comments. Let's not worry about solutions yet - we'll get to that. 
+
+- @stebalien's 'Not blocked by steb' issue. Read it over - there's a new label, and new informaiton about how to reach Steven when you need him.
+
+#### Discussion
+- Datastore benchmarking update from @magik6k
+  - beautiful beautiful graphs that tell us lots more information
+  - would love to see them with an additional 0 for size
+  - jfs is broken
+  - using the default driver (ntfs doesn't have great performance over time)
+  - interested to see graphs with different data sets (expect to see this behave differently on different configurations of data - many small files, large files, etc)
+  - plan: expand to run more benchmarks on more computers to integrate with AWS to auto-run more stuff there.
+
+### Week Update
+
+@name
+  - Done:
+  - Blocked:
+  - Next:
+
+@marten-seemann
+ - Done: 
+  * a lot of review of PRs for the QUIC spec
+  * more work on draft-17
+  * started working on a benchmarking suite for quic-go
+ - Blocked: ---
+ - Next:
+ 	* more draft-17 work, especially header encryption and stateless resets
+
+@djdv
+ - Blocked:
+    - still OOO ⚖️
+
+@schomatis (not attending)
+ - Done/In Progress:
+   - Improving content for a concept document about files in IPFS (https://github.com/ipfs/docs/issues/133).
+   - Write mid-level document on the `ipfs files` API (https://github.com/ipfs/go-ipfs/pull/5052).
+   - With the DAG walker merged finish up the DAG reader refactor using it (https://github.com/ipfs/go-unixfs/pull/12).
+- Next:
+   - MFS interface (https://github.com/ipfs/go-mfs/issues/3).
+
+@aschmahmann
+ - Done/In Progress:
+  - Decided to push off more sophisticated synchronization of version graphs for now
+    - Will revisit after a chat demo is working
```

Comment:
@aschmahmann Shouldn't the chat app always work (given that bitswap exists) and then with your graphsync version get faster?

Reply:
Unfortunately, there's a bit of a nomenclature issue with GraphSync (there are currently 3 proposals all called GraphSync). My understanding is that GraphSync is about asking for DAGs of files instead of one by one asking for files corresponding to CIDs.

The synchronization of version graphs I'm referring to is that if Alice has a DAG `A` indicating the versions and branches of a data structure (e.g. a Git repo) and Bob has a DAG `B` indicating the versions and branches of that same data structure, there should be a protocol in place for Alice and Bob to communicate and each end up with the resulting DAG `C` which is the union of `A` and `B` . This may use some of the same techniques we end up using in one of the GraphSync proposals, but so far neither solution seems to fully contain the other. In order to try and be efficient I'm planning on pushing off more complex implementations of version graph synchronization (and just using a simple graph merge that sends the whole graph over the network) until I see more of how GraphSync develops and if there are any techniques I can borrow from there.

It is currently fairly straightforward to have a chat application work over pubsub. However, having this structure happens to make it very easy to implement at causal chat application (i.e. a user will not be able to see a message without first having seen the messages that proceeded it).

I'll try and document this more thoroughly, but please let me know if you have any questions.